python/pyarrow/tests/test_parquet.py

```diff
@@ -166,7 +197,12 @@ def test_pandas_parquet_2_0_roundtrip(tempdir, chunk_size):
 
     _write_table(arrow_table, filename, version="2.0",
                  coerce_timestamps='ms', chunk_size=chunk_size)
-    table_read = pq.read_pandas(filename)
+    # TODO(datasets)
+    use_threads = (
+        False if not use_legacy_dataset and chunk_size is not None else True)
```

Comment:
maybe this would be more clear if you extracted
```
def deterministic_row_order(use_legacy_dataset, chunk_size=None):
    return False if not use_legacy_dataset and chunk_size is not None else True
```